tyxe/util.py

```diff
@@ -9,18 +9,11 @@
 import pyro.infer.autoguide.guides
 import pyro.nn.module as pyromodule
 
-
-def deep_hasattr(obj, name):
-    try:
-        pyro.util.deep_getattr(obj, name)
-        return True
-    except AttributeError:
-        return False
-
-
-deep_setattr = pyro.infer.autoguide.guides._deep_setattr
-deep_getattr = pyro.util.deep_getattr
-
+try:
```

Comment:
If only one version of pyro is to be supported, then this can be simplified.

Reply:
Yes I'd go with only supporting the most recent pyro version. We could just wrap their `deep_attr` function with a deprecation warning (see [here](https://github.com/pyro-ppl/pyro/blob/d7687ae0f738bd81a792dabbb18a53c0fce73765/pyro/generic.py#L9) for an example from the pyro codebase) and eventually remove them here alltogether.